frontend/controller/actions/group.js

```diff
@@ -907,6 +912,72 @@ export default (sbp('sbp/selectors/register', {
       sbp('okTurtles.events/emit', REPLACE_MODAL, 'IncomeDetails')
     }
   },
+  'gi.actions/group/fixAnyoneCanJoinLink': async function ({ contractID }) {
+    const state = await sbp('chelonia/contract/state', contractID)
+    const now = await sbp('chelonia/time')
+
+    // For used invites, we add all 'anyone can join' invites and do the following:
+    //  1. For non-expired invites, we consider all of the invites
+    //       (initialQuantity) as used
+    //  2. For expired invites, we consider the quantity actually used
+    //  3. We add up 1. and 2. above
+    // Then, we take the difference and, if the number is less than
+    // MAX_GROUP_MEMBER_COUNT, we create a new invite.
```

Comment:
This comment doesn't make sense to me for the following reasons:

> we add all 'anyone can join' invites

This is plural - but there is only one anyone-can-join invite when this code is run. This code should never be run more than once - by postUpgradeVerification, to fix the current situation we're in. And in the current situation, there can only ever be a single anyone-can-join invite.

> 1. For non-expired invites, we consider all of the invites (initialQuantity) as used as used
> 2. For expired invites, we consider the quantity actually used

We should not be touching or caring about expired (or non-expired) invites that are not the anyone-can-join invite. Those should be left alone completely.

And for the anyone-can-join invite, whether or not it is expired is irrelevant. We want to replace the *single* anyone-can-join invite with a different, single, anyone-can-join invite (and pretend like the original one never existed in the first place).

Then we take the number of used invites, and subtract that from `MAX_GROUP_MEMBER_COUNT`, and that is the amount we have remaining.

Reply:
> We should not be touching or caring about expired (or non-expired) invites that are not the anyone-can-join invite. Those should be left alone completely.

That is exactly what this is doing.

> This is plural - but there is only one anyone-can-join invite when this code is run. 

Right now, there can be multiple anyone-can-join invites.

Reply:
> Right now, there can be multiple anyone-can-join invites.

Please elaborate how this is possible when the app only creates one?

Reply:
The app creates multiple.

For example, if you created a group with 60 invites (1 of which was used and then expired), now you'll have 2 invites, one expired, and one with 149 slots that doesn't expire.

For example, if you created a group with 60 invites (1 of which was used and hasn't yet expired), now you'll have 2 invites, one non-expired with 59 remaining, and one with 91 slots that doesn't expire.

Reply:
> For example, if you created a group with 60 invites (1 of which was used and then expired), now you'll have 2 invite links, one expired, and one with 149 slots that doesn't expire.

But this code is for getting rid of the 60-max invite and replacing it with the 150-max invite, isn't it? At least that's what it's supposed to be doing.

> For example, if you created a group with 60 invites (1 of which was used and hasn't yet expired), now you'll have 2 invite links, one non-expired with 59 remaining, and one with 91 slots that doesn't expire.

That's not what this PR is supposed to do. The PR is supposed to get rid of the 60-max invite and replace it with a 150-max invite (subtracting however many from the 60-max were used).

So no, you shouldn't have 2 invite links. And in this example, a new non-expiring anyone-can-join invite should be created with 149 invites remaining.

Reply:
I see, but doing it the way you suggest will break invite links that were already shared.

Reply:
That's OK, they can be re-shared, and we can remember to announce that when we release the update.

Reply:
In any case, the math here is still (almost) correct because you could have multiple invite links when using, e.g., an alternate client, so you need to add all of them up.

Reply:
In fact, if you'd like to update this PR with a new entry for the `NewsAndUpdates.vue` file, please feel free to do so and add this text:

```js
  {
    createdAt: new Date('2024-10-29'),
    title: '1.1.0: Anyone-can-join Invites Updated! Please read!',
    content: "Version 1.1.0 introduces many new features and bug fixes, and among them is replacing the old anyone-can-join invite links with new ones that will let your group grow to Dunbar's Number. \n\n" +
      '**The old links are expired, please use the new ones!**'
  },
```